### ISSUE TITLE
Usage of mobile data while having WIFI-only sync

### ISSUE THREAD

**Issue by erikvanoosten** [syncthing/syncthing-android#722]
Yesterday (2016-08-13) and the day before, Syncthing used all mobile data I had available (12 GB). This is not something I expected because I have enabled WIFI-only sync.

I do not know how to reproduce this.

### Version Information

    App Version: 0.8.1
    Syncthing Version: v0.14.3
    This version is installed since 2016-08-01
    Android Version: Android 6.0.1
    Model: Google Nexus 5

**Comment by Nutomic** [comment]
Logs would be be really useful. Without logs or a way to reproduce, there's no way we can fix this.

**Comment by erikvanoosten** [comment]
Where can I find the logs?

**Comment by erikvanoosten** [comment]
The two logs I could get via the settings menu are too recent. The first entry in both are just a few hours old.

**Comment by erikvanoosten** [comment]
There is something I noticed and which I find suspicious. When I connect to a network, syncthing-android would start running immediately, even though I did not actually finish the log-in screen yet. The Android OS knows this because the WIFI icon still has an exclamation mark in this stage.

**Comment by capi** [comment]
Then there seems to be yet another difference in your ROM (seems to be the Google one) to other ROMs :-( Others are complaining that it is not starting, because it thinks there is no Internet connection. Gotta love the diversity of Android :-/

**Comment by Nutomic** [comment]
Which login screen do you mean? The wifi password dialog? Or some web-based login to access the internet?

From other issues, it seems that Android won't trigger a "wifi connected" event until it can actually connect to the internet (as mentioned by @capi). So this sounds weird, especially on vanilla Android.

**Comment by jfwjfw** [comment]
I have noticed this problem on an Nexus 5X and LG G3 (the only Android phones I use it syncthing on. With both phones it appears to reverse wifi detection - ie if I set syncthing to sync only on wifi, it syncs when no wifi and won't sync on wifi.

This problem was evident in July, was fixed, then came back again a few weeks ago (not sure exactly when).

I think #700, #625, #630 are likely to be the same bug.

**Comment by Nutomic** [comment]
@jfwjfw Is there any way you can reliably reproduce this behaviour? Logs might also be helpful.

**Comment by Nutomic** [comment]
Is that really an issue? It might be unexpected, but there's nothing we can do. Android decides when it sends us the "connected to wifi" intent.

**Comment by lkwg82** [comment]
@nutomic Agree! As far as I know Android has the feature to switch smoothly to mobile/wifi on connection availibility.

**Comment by erikvanoosten** [comment]
Can someone please label this as `invalid`?
